casper/src/main/scala/coop/rchain/casper/util/comm/CasperPacketHandler.scala

```diff
@@ -0,0 +1,496 @@
+                                         shardId)
+                     _   <- MultiParentCasperRef[F].set(casper)
+                     _   = println(s"Set")
+                     abh = new ApprovedBlockReceivedHandler[F](casper, approvedBlock)
+                     _   <- capserHandlerInternal.set(abh)
+                   } yield ()
+               }
+      } yield cont
+  }
+
+  /** Node in this state will query peers in the network with [[ApprovedBlockRequest]] message
+    * and will wait for the [[ApprovedBlock]] message to arrive. Until then  it will respond with
+    * `F[None]` to all other message types.
+    **/
+  private[comm] class BootstrapCasperHandler[
+      F[_]: Sync: Capture: ConnectionsCell: NodeDiscovery: BlockStore: TransportLayer: Log: Time: ErrorHandler: SafetyOracle: RPConfAsk: LastApprovedBlock](
+      runtimeManager: RuntimeManager,
+      shardId: String,
+      validatorId: Option[ValidatorIdentity],
+      validators: Set[ByteString])(implicit scheduler: Scheduler)
+      extends CasperPacketHandlerInternal[F] {
+    private val nonePacket: F[Option[Packet]] = Applicative[F].pure(None: Option[Packet])
+
+    override def handleBlockMessage(bm: BlockMessage): F[Option[Packet]] = nonePacket
+    override def handleBlockRequest(peer: PeerNode, br: BlockRequest): F[Option[Packet]] =
+      nonePacket
+    override def handleApprovedBlockRequest(peer: PeerNode,
+                                            br: ApprovedBlockRequest): F[Option[Packet]] =
+      Option(noApprovedBlockAvailable(peer)).pure[F]
+
+    override def handleUnapprovedBlock(peer: PeerNode, ub: UnapprovedBlock): F[Option[Packet]] =
+      nonePacket
+    override def handleBlockApproval(ba: BlockApproval): F[Option[Packet]] = nonePacket
+
+    override def handleApprovedBlock(
+        ab: ApprovedBlock): F[(Option[MultiParentCasper[F]], Option[Packet])] =
+      for {
+        isValid <- Validate.approvedBlock[F](ab, validators)
+        casper <- if (isValid) {
+                   for {
+                     _            <- Log[F].info("CASPER: Valid ApprovedBlock received!")
+                     blockMessage = ab.candidate.flatMap(_.block).get
+                     _            <- BlockStore[F].put(blockMessage.blockHash, blockMessage)
+                     internalMap  <- BlockStore[F].asMap()
+                     _            <- LastApprovedBlock[F].set(ab)
+                     casper = MultiParentCasper
+                       .hashSetCasper[F](runtimeManager,
+                                         validatorId,
+                                         blockMessage,
+                                         internalMap,
+                                         shardId)
+                   } yield Option(casper)
+                 } else
+                   Log[F]
+                     .info("CASPER: Invalid ApprovedBlock received; refusing to add.")
+                     .map(_ => none[MultiParentCasper[F]])
```

Comment:
Can we be consistent on the parens?

```
if {

} else {

}
```

Also I prefer

```
for {
   _ <-                    Log[F]
                     .info("CASPER: Invalid ApprovedBlock received; refusing to add.")
} yield none[MultiParentCasper[F]]
```
to be consistent with the above.

Reply:
IMO it's an overkill and `for comprehensions` after normalizing will end up with additional `map` statement.